win32ss/printing/base/winspool/printers.c

```diff
@@ -17,6 +17,92 @@
 static const WCHAR wszWindowsKey[] = L"Software\\Microsoft\\Windows NT\\CurrentVersion\\Windows";
 static const WCHAR wszDeviceValue[] = L"Device";
 
+BOOL strW2A(PWSTR wstring, PSTR pstring)
```

Comment:
Several things wrong with this function:
* The logic only works for in-place conversions of Unicode to ANSI. As such, `wstring` and `pstring` need to point to the same address. As that is a requirement, the function should only take 1 parameter anyway.
* TRUE for failure and FALSE for success is totally counterintuitive. Please reverse that.
* If that function is going to be used by multiple files, it should be put into a new file, e.g. "utils.c", with an `extern` line added to precomp.h.
* Function and variable naming should resemble the existing syntax and again emphasize that this function can only be used for in-place conversions, e.g. `BOOL UnicodeToAnsiInPlace(PWSTR pwszField)`
* I have only used `goto Cleanup` whenever I needed a single exit path to free memory allocated at multiple positions. This function does not need that and due to its defined purpose, it likely won't do in the future. So it's way more readable to use no `goto` here.

Reply:
@ColinFinck, Thank you for your review and comments. I believe I have taken care of all of them. Let me know of anything else that I need to change. Thanks.